source/github-helpers/index.ts

```diff
@@ -34,9 +34,33 @@ export const getCurrentCommittish = (pathname = location.pathname, title = docum
 		return parsedTitle.groups!.branch;
 	}
 
+	// .last needed for #2799 GHE #4187
+	const feedLink = select.last('link[type="application/atom+xml"]');
+	if (feedLink) {
+		return new URL(feedLink.href)
+			.pathname
+			.split('/')
+			.slice(4) // Drops the initial /user/repo/route/ part
+			.join('/')
+			.replace(/\.atom$/, '');
+	}
+
 	return unslashedCommittish;
 };
 
+export const getCurrentBranchFromFindLink = async (): Promise<string | undefined> => {
```

Comment:
```suggestion
export const getCurrentCommittishAnywhere = async (): Promise<string | undefined> => {
    const committish = getCurrentCommittish();
    if (committish) {
        return committish;
    }

```

This way, in places like more-dropdown, you only need to call this function + getCurrentBranch. The only reason for this function to exist at all outside getCurrentCommittish is that that one is domready safe and this one isn’t.